skyportal/models.py

```diff
@@ -1758,6 +1759,231 @@ def api_class(self):
 
+    fields = relationship("InstrumentField")
+    tiles = relationship("InstrumentFieldTile")
+    plans = relationship("ObservingPlan")
+
+
+class InstrumentField(ha.Region, Base):
+
+    tile_class = lambda: InstrumentFieldTile  # pylint: disable=E731 # noqa
+
+    instrument_id = sa.Column(
+        sa.ForeignKey('instruments.id', ondelete="CASCADE"),
+        nullable=False,
+        doc='Instrument ID',
+    )
+
+    instrument = relationship(
+        "Instrument",
+        foreign_keys=instrument_id,
+        doc="The Instrument that this field belongs to",
+    )
+
+    field_id = sa.Column(
+        sa.Integer,
+        index=True,
+        doc='The Field ID for the tile.',
+        nullable=False,
+    )
+
+    tiles = relationship("InstrumentFieldTile")
+
+
+class InstrumentFieldTile(ha.Tile, Base):
+
+    instrument_id = sa.Column(
+        sa.ForeignKey('instruments.id', ondelete="CASCADE"),
+        nullable=False,
+        doc='Instrument ID',
+    )
+
+    instrument = relationship(
+        "Instrument",
+        foreign_keys=instrument_id,
+        doc="The Instrument that this tile belongs to",
+    )
+
+    instrument_field_id = sa.Column(
+        sa.ForeignKey('instrumentfields.id', ondelete="CASCADE"),
+        nullable=False,
+        doc='Instrument Field ID',
+    )
+
+    field = relationship(
+        "InstrumentField",
+        foreign_keys=instrument_field_id,
+        doc="The Field that this tile belongs to",
+    )
+
+
+class ObservingPlan(Base):
+    """Tiling information, including the event time, localization ID, tile IDs,
+    and plan name"""
+
+    instrument_id = sa.Column(
+        sa.ForeignKey('instruments.id', ondelete="CASCADE"),
+        nullable=False,
+        doc='Instrument ID',
+    )
+
+    dateobs = sa.Column(
+        sa.ForeignKey('gcnevents.dateobs', ondelete="CASCADE"),
+        nullable=False,
+        index=True,
+        doc='UTC event timestamp',
+    )
+
+    plan_name = sa.Column(sa.String, primary_key=True, unique=True, doc='Plan name')
+
+    validity_window_start = sa.Column(
+        sa.DateTime,
+        nullable=False,
+        default=lambda: datetime.now(),
+        doc='Start of validity window',
+    )
+
+    validity_window_end = sa.Column(
+        sa.DateTime,
+        nullable=False,
+        default=lambda: datetime.now() + timedelta(1),
+        doc='End of validity window',
+    )
+
+    plan_args = sa.Column(sa.JSON, nullable=False, doc='Plan arguments')
+
+    class Status(enum.IntEnum):
+        WORKING = 0
+        READY = 1
+        SUBMITTED = 2
+
+    status = sa.Column(
+        sa.Enum(Status), default=Status.WORKING, nullable=False, doc='Plan status'
+    )
+
+    planned_observations = relationship("PlannedObservation")
+
+    @property
+    def start_observation(self):
+        """Time of the first planned observation."""
+        if self.planned_observations:
+            return self.planned_observations[0].obstime
```

Comment:
shouldn't we be sorting planned observations by `obstime` to find the earliest one? or will they all have same `obstime`?

Reply:
oh yeah!